src/core.c/Proc/Async.rakumod

```diff
@@ -391,7 +391,10 @@ my class Proc::Async {
         });
 
         nqp::bindkey($callbacks, 'error', -> Mu \err {
-            my $error := X::OS.new(os-error => err);
+            my %details = os-error => err;
+            my $code = try (err ~~ /"(error code " ("-"? \d+) ")"/)[0].Int;
```

Comment:
Seems to me that it'd be very easy to pass the error code as a separate argument to the `error` callback which would save us the potentially flaky parsing of the error message.